src/WebAppDIRAC/WebApp/handler/SystemAdministrationHandler.py

```diff
@@ -636,101 +633,30 @@ def web_getSelectionData(self):
     @asyncGen
     def web_ComponentLocation(self):
 
-        _setup = self.getUserSetup()
-        setup = _setup.split("-")[-1]
-
-        hosts = []
-        result = ComponentMonitoringClient().getHosts({}, False, False)
-        if result["OK"]:
-            hosts = [h["HostName"] for h in result["Value"] if h.get("HostName")]
-
-        componentTypes = ["Services", "Agents"]
+        componentTypes = ["Service", "Agent", "Executor"]
         componentTypes = self.get_arguments("ComponentType") or componentTypes
 
-        componentNames = []
-        if "ComponentName" in self.request.arguments:
-            componentNames = list(json.loads(self.get_argument("ComponentName")))
-
         componentModules = []
         if "ComponentModule" in self.request.arguments:
             componentModules = list(json.loads(self.get_argument("ComponentModule")))
 
-        showAll = 0
-        if "showAll" in self.request.arguments:
-            showAll = int(self.get_argument("showAll"))
-
         selectedHosts = []
         if "Hosts" in self.request.arguments:  # we only use the selected host(s)
             selectedHosts = list(json.loads(self.get_argument("Hosts")))
-        retVal = gConfig.getSections("/Systems")
-
-        compMatching = {}
-        fullNames = []
-        if retVal["OK"]:
-            systems = retVal["Value"]
-            for i in systems:
-                for compType in componentTypes:
-                    compPath = "/Systems/%s/%s/%s" % (i, setup, compType)
-                    retVal = gConfig.getSections(compPath)
-                    if retVal["OK"]:
-                        components = retVal["Value"]
-                        for j in components:
-                            path = "%s/%s" % (i, j)
-                            if j in componentNames:
-                                fullNames += [path]
-                                compMatching[path] = path
-                            modulepath = "%s/%s/Module" % (compPath, j)
-                            module = gConfig.getValue(modulepath, "")
-                            if module != "" and module in componentModules:
-                                fullNames += [path]
-                            elif module == "" and j in componentModules:
-                                fullNames += [path]
 
-                            compMatching[path] = module if module != "" else path
+        condDict = {"Type": componentTypes, "DIRACModule": componentModules}
+        gLogger.debug("condDict" + str(condDict))
 
+        res = ComponentMonitoringClient.getComponents(condDict, True, True)
+        if not res["OK"]:
+            self.finish({"success": "false", "error": res["Message"]})
```

Comment:
If you do not return at this point, the code will continue to run until the end of the function and will fail with `KeyError: 'Value'`

```suggestion
            self.finish({"success": "false", "error": res["Message"]})
            return
```